core/services/postgres/advisory_lock.go

```diff
@@ -0,0 +1,119 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"sync"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// NOTE: All advisory lock class IDs used by the Chainlink application MUST be
+// kept here to avoid accidental re-use
+const (
+	AdvisoryLockClassID_EthBroadcaster int32 = 0
+	AdvisoryLockClassID_JobSpawner     int32 = 1
+	AdvisoryLockClassID_EthConfirmer   int32 = 2
+
+	// ORM takes lock on 1027321974924625846 which splits into ClassID 239192036, ObjID 2840971190
```

Comment:
Meaningless, I believe it was originally a randomly generated number.

PG advisory locks are actually always two 32-bit integers. If you give one 64-bit integer it will do some sort of bitshifting to break it into two 32-bit integers internally.

These two 32-bit ints are the numbers that postgres uses internally for this 64-bit lock ID. Found by actually querying the postgres lock table while the application advisory lock is being held.

Reply:
Followup to my comment above, I was mistaken.

The two numbers are actually nonsense :). Bigint keys operate on a completely different scope than the two 32-bit keys.

Reply:
This comment should be removed.

Reply:
Let's do it when we do the next batch of work on the advisory lock (i.e. converting `orm.PostgresLockingStrategy` to use `postgres.AdvisoryLock`